tests/integration/pkg/hooks/hooks.go

```diff
@@ -215,3 +226,57 @@ func createApiGatewayCRObjectFromTemplate(name string) (v1alpha1.APIGateway, err
 	apiGateway.Name = name
 	return apiGateway, nil
 }
+
+func deleteBlockingResources(ctx context.Context) error {
```

Comment:
I think a more generic approach, like it's in Istio would be better here. We add resources to a list if they are created during a test and they'll be deleted in the after hook.
I think we didn't implement it that way to keep it simpler for now and adapt it when we rewrite this integration tests?

Reply:
Yes, agree, I like the controlled approach in Istio. This is done just for the sake of moving the urgent API-Gateway impl tasks faster for the time being. We should invest more time into refactoring / making better the integration tests as a follow-up.